qemu/tests/block_disk_not_default_options.py

```diff
@@ -0,0 +1,82 @@
+from virttest import error_context, qemu_qtree
+
+from provider import disk_utils
+from provider.block_devices_plug import BlockDevicesPlug
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test the disk non-default options.
+
+    Steps:
+        1. Boot a VM with packed=true and page-per-vq=true
+        2. Verify the packed and page-per-vq values
+        3. Do some basic I/O operation in the VM
+        4. Unplug and hotplug the disk
+        5. Do again a basic I/O operation
+
+    :param test: QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env: Dictionary with test environment.
+    """
+
+    def verify_non_default_options(option_key, expected_value):
+        """Verify the non-default values.
+        :param option_key: the QEMU option to be checked.
+        :param expected_value: the QEMU option expected value.
+        """
+        error_context.context(f"Verify {option_key} from monitor.", test.log.info)
+        qtree = qemu_qtree.QtreeContainer()
+        qtree_option_value = False
+        try:
+            qtree.parse_info_qtree(vm.monitor.info("qtree"))
+        except AttributeError:
+            test.cancel("Monitor doesn't support qtree skip this test")
+        error_msg = (
+            f"'{option_key}' value mismatch: expected %s but report from monitor is: %s"
+        )
+
+        for node in qtree.get_nodes():
+            if (
+                isinstance(node, qemu_qtree.QtreeDev)
+                and node.qtree.get(option_key) == expected_value
+            ):
+                qtree_option_value = node.qtree.get(option_key)
+                error_context.context(
+                    f"The qtree_option_value: {qtree_option_value}", test.log.debug
+                )
+
+        if qtree_option_value != expected_value:
+            error_msg = error_msg % (str(expected_value), qtree_option_value)
+            test.fail(error_msg)
+
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    session = vm.wait_for_login(timeout=360)
+
+    qtree_check_images = params.get("qtree_check_images", "stg0").split()
+
+    tmp_dir = params.get("tmp_dir", "/var/tmp/test")
+    disk_utils.execute_dd_on_disk_by_id(
+        params, qtree_check_images[0], session, tmp_dir, test
+    )
+
+    for image in qtree_check_images:
+        qtree_check_image = eval(params.get(f"qtree_check_{image}", "{}"))
```

Comment:
`eval` is not a good choice, reading your cfg, you may want convert json to dict?

Reply:
@PaulYuuu 
Yes, I think so, it's been a long time since I push this code... :sweat_smile:

Reply:
@PaulYuuu got it, thanks for the review I am gonna give a try !

Reply:
No, it seems it's not a JSON, it's like a properties file, after applying `get_dict`:
`ValueError: failed to find "=" in "{"packed"" (value for qtree_check_stg0)`

Reply:
I will keep the current implementation, thanks @PaulYuuu for the review

Reply:
You should check the usage of get_dict from utils_params.py. it requests like `key = foo1=bar1 foo2=bar2`

Reply:
What you see is what you get is better. why do we need to convert to dict instead of direct dict expression?
Anyway, the current code is acceptable to me.